cajita/src/Cajita_DynamicPartitioner.hpp

```diff
@@ -903,6 +762,225 @@ class SparseDimPartitioner : public BlockPartitioner<NumSpaceDim>
             ( global_cells_per_dim[2] >> cell_bits_per_tile_dim ) + 1 );
     }
 };
+
+/*!
+  Dynamic mesh block partitioner. (Current Version: Support 3D only) Workload
+  are computed from particle distribution.
+
+  \tparam Device Kokkos device type.
+  \tparam CellPerTileDim Cells per tile per dimension.
+  \tparam NumSpaceDim Dimemsion (The current version support 3D only)
+*/
+template <typename Device, unsigned long long CellPerTileDim = 4,
+          std::size_t NumSpaceDim = 3>
+class ParticleDynamicPartitioner
+    : public DynamicPartitioner<Device, CellPerTileDim, NumSpaceDim>
+{
+    using base = DynamicPartitioner<Device, CellPerTileDim, NumSpaceDim>;
+
+  protected:
+    using base::_workload_per_tile;
+
+  public:
+    using base::cell_bits_per_tile_dim;
+    using base::num_space_dim;
+    using typename base::execution_space;
+
+    /*!
+      \brief Constructor - automatically compute ranks_per_dim from MPI
+      communicator
+      \param comm MPI communicator to decide the rank nums in each dimension
+      \param max_workload_coeff threshold factor for re-partition
+      \param workload_num total workload(particle/tile) number, used to compute
+      workload_threshold
+      \param num_step_rebalance the simulation step number after which one
+      should check if repartition is needed
+      \param global_cells_per_dim 3D array, global cells in each dimension
+      \param max_optimize_iteration max iteration number to run the optimization
+    */
+    ParticleDynamicPartitioner(
```

Comment:
Can simplify with just `using base::base;` since we're not doing anything except calling the base constructor